graphql_compiler/compiler/metadata.py

```diff
@@ -154,11 +161,20 @@ def get_location_info(self, location):
                                  u'{}'.format(location))
         return location_info
 
+    @property
+    def tags(self):
+        """Return a dict of tag name to TagInfo for the query."""
+        return self._tags
```

Comment:
Dicts are mutable, let's not return references to mutable objects which the caller may abuse. Look at how the `registered_locations` property gets around this problem.